.github/workflows/build.yml

```
@@ -0,0 +1,43 @@
+# The present workflow was made based on the following references:
+# - https://github.com/actions/cache/blob/main/examples.md#haskell---cabal
+# - https://github.com/haskell/time/blob/master/.github/workflows/ci.yml
+# - https://github.com/stackbuilders/stache/blob/master/.github/workflows/ci.yaml
+# - https://markkarpov.com/post/github-actions-for-haskell-ci.html
+---
+name: Build
+
+on: push
+
+concurrency:
+  group: build-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  haskell:
+    runs-on: ${{ matrix.os }}
```

@cptrodolfox it looks like all different GHC versions are tested against `ubuntu-latest` in that case, I would suggest getting rid of a level of indirection `${{ matrix.os }}` and hard-code the value directly.

```suggestion
    runs-on: ubuntu-latest
```